_posts/2025-02-24-ptpc-fp8-rocm.md

```diff
@@ -0,0 +1,295 @@
+
+* **What's new?** [PTPC-FP8 quantization](https://github.com/vllm-project/vllm/pull/12501) is now supported in vLLM (v0.7.3+) on AMD ROCm.  
+* **Why is it good?** You get speeds similar to other FP8 methods, but with accuracy much closer to the original (BF16) model quality. It's the best FP8 option for ROCm.  
+* **How to use it:**  
+  1. Install ROCm.  
+  2. Get the latest vLLM (v0.7.3 or newer).  
+  3. Add the `--quantization ptpc_fp8` flag when running your Hugging Face model. No need to pre-quantize!
+
+
+<img align="center" src="/assets/figures/ptpc/PTPC121.png" alt="What is PTPC-FP8" width="90%" height="90%">
+
+**What is PTPC-FP8?** It's a method for FP8 weights *and* activations quantization. It uses per-token scaling for activations and per-channel scaling for weights, giving you better accuracy than traditional per-tensor FP8.
+
+## Introduction
+
+Large Language Models (LLMs) are revolutionizing how we interact with technology, but their immense computational demands can be a barrier. What if you could run these powerful models faster and more efficiently on your AMD GPUs, without sacrificing accuracy? Now you can! This post introduces a breakthrough: PTPC-FP8 quantization in vLLM, optimized for AMD's ROCm platform. Get ready for near-BF16 accuracy at FP8 speeds, directly using Hugging Face models – no pre-quantization needed! We'll show you how it works, benchmark its performance, and get you started.
+
+### The Challenge of LLM Quantization and the PTPC-FP8 Solution
+
+Running large language models is computationally expensive. FP8 (8-bit floating-point) offers a compelling solution by reducing memory footprint and accelerating matrix multiplications, but traditional quantization approaches face a critical challenge with LLMs.
+
+#### The Outlier Problem
+
+LLMs develop activation outliers as they scale beyond certain sizes. These unusually large values create significant quantization challenges:
+
+- Most values receive few effective bits of precision when using per-tensor quantization
+- Outliers appear persistently in specific channels across different tokens
+- While weights are relatively uniform and easy to quantize, activations are not
+
+#### PTPC: A Precision-Targeted Approach
+
+PTPC-FP8 (Per-Token-Activation, Per-Channel-Weight FP8) addresses this challenge by using tailored scaling factors based on three key observations:
+
+1. Outliers consistently appear in the same channels
+2. Channel magnitudes within a token vary widely
+3. The same channel's magnitude across different tokens remains relatively stable
+
+This insight led to a dual-granularity approach:
+* **Per-Token Activation Quantization**: Each input token receives its own scaling factor
+* **Per-Channel Weight Quantization**: Each weight column gets a unique scaling factor
+
+<img align="right" src="/assets/figures/ptpc/PTPC-Diagram.png" alt="Per-Token Activation + Per-Channel Weight Quantization" width="50%" height="50%">
```

Comment:
This diagram is unreadable on mobile:

![image](https://github.com/user-attachments/assets/3bebbcf5-de84-4877-a898-437f39e96f1b)